### PR TITLE
Adding time threshold compatibility to MTDThresholdClusterizer

### PR DESCRIPTION
**Pull request by pmeridian**
This PR improves the MTD clusters @PU200 introducing a configurable time compatibility threshold (now set to a very loose @10sigma of the sigma of difference) between the seed and the hits to be attached to the cluster. It improves significantly the purity of the hits attached to the clusters as can be seen here:

Energy PU comparison in BTL with new configuration (Single MU)
btlbestcluster_energy_mupucomp-new

Old configuration
btlbestcluster_energy_mupucomp-old

Cluster size and time shows a similar behaviour:
btlbestcluster_time_mupucomp
btlbestcluster_size_mupucomp
etlbestcluster_size_mupucomp

Efficiency of the best hit matching is not affected:
divide_btlmatchedbestclustertrack_eta_by_btltrack_eta_mupuoldcomp

Important to be used in the next mtd release
@lgray @bendavid  @fabiocos  @franzoni

### CODE REVIEW

File: RecoLocalFastTime/FTLClusterizer/src/MTDThresholdClusterizer.cc [hunk 1]
```diff
@@ -256,6 +258,8 @@ MTDThresholdClusterizer::make_cluster( const FTLCluster::FTLHitPos& hit )
       for ( auto c = std::max(0,int(acluster.y[curInd]-1)); c < std::min(int(acluster.y[curInd]+2),int(theBuffer.columns())) && !stopClus; ++c) {
 	  for ( auto r = std::max(0,int(acluster.x[curInd]-1)); r < std::min(int(acluster.x[curInd]+2),int(theBuffer.rows()))  && !stopClus; ++r)  {
 	  if ( theBuffer.energy(r,c) > theHitThreshold) {
+	    if (fabs(theBuffer.time(r,c) - seed_time) > theTimeThreshold*sqrt( theBuffer.time_error(r,c)*theBuffer.time_error(r,c) + seed_time_error*seed_time_error))
```
Comment (slava77) [inline RecoLocalFastTime/FTLClusterizer/src/MTDThresholdClusterizer.cc]: std::abs

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by fabiocos** [comment]
please test workflow 27434.0,27834.0

File: RecoLocalFastTime/FTLClusterizer/plugins/MTDClusterProducer.cc [hunk 2]
```diff
@@ -109,6 +109,7 @@ MTDClusterProducer::fillDescriptions(edm::ConfigurationDescriptions& description
   desc.add<double>("HitThreshold", 0.);
   desc.add<double>("SeedThreshold", 0.);
   desc.add<double>("ClusterThreshold", 0.);
+  desc.add<double>("TimeThreshold", 10.);
```
Comment (slava77) [inline RecoLocalFastTime/FTLClusterizer/plugins/MTDClusterProducer.cc]: This seems like a leftover/incomplete code review from #25063 

 the 4 lines above should be replaced by a single call
```MTDThresholdClusterizer::fillDescriptions(desc);```

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/25644	027f20e
- code changes are in line with the PR description and the follow up review
- jenkins tests pass and comparisons with the baseline show small changes in the MTD clusters, tracking hits, and MTD-extended tracks somewhat in line with the much higher stat plots posted with the PR description.

**Comment by kpedro88** [comment]
+upgrade

**Comment by fabiocos** [comment]
+1

**Pull request merged**